homeassistant/components/simplisafe/__init__.py

```diff
@@ -539,17 +538,25 @@ async def update_system(system):
         for result in results:
             if isinstance(result, InvalidCredentialsError):
                 if self._emergency_refresh_token_used:
-                    LOGGER.error(
-                        "Token disconnected or invalid. Please re-auth the "
-                        "SimpliSafe integration in HASS"
-                    )
-                    self._hass.async_create_task(
-                        self._hass.config_entries.flow.async_init(
-                            DOMAIN,
-                            context={"source": "reauth"},
-                            data=self._config_entry.data,
+                    matching_flows = [
+                        flow
+                        for flow in self._hass.config_entries.flow.async_progress()
+                        if flow["context"]["source"] == SOURCE_REAUTH
+                        and flow["context"]["unique_id"] == self._config_entry.unique_id
```

Comment:
Don't assume these values exist
```suggestion
                        if flow["context"].get("source") == SOURCE_REAUTH
                        and flow["context"].get("unique_id") == self._config_entry.unique_id
```